Yoast/wordpress-seo
# Introduces Yoast indexables and ORM

## Summary

This PR can be summarized in the following changelog entry:

* Introduces Indexables. Creates tables and introduces an ORM to access these tables. This functionality is only available on servers running PHP 5.3 or higher.

## Relevant technical choices:

* Wrapped the code in `namespace` conditional checking, to make sure we are in PHP 5.3
* Added 5.3 specific tests in a separate folder.
* Added migrations scripts

## Test instructions

_**Note:** This is a huge feature, I want to apologize for possibly unclear or missing test instructions. If you have questions please ping me and I will help you walk through acceptance._ 

This PR can be tested by following these steps:
Add `define( 'YOAST_SEO_INDEXABLES', true );` to the `wp-config` file.

The indexables are applicable for posttypes like `post`, `page`, `custom post types`, taxonomies like `categories` `tags`, `custom taxonomies` and `authors`. To get a full covered test it is importing to test these situations for every single one of them. Data should be saved in the database:
1. Create a new item (for taxonomies nothing should be saved, because there is no metabox there).
1. Edit an existing item
1. Delete an existing item

### What must be verified and where?
1. Go to phpmyadmin and login. 
1. We have a couple of database tables: one is migrations (`wp_yoast_migrations`) , one is for indexables (`wp_yoast_indexable`), one is for primary terms (`wp_yoast_primary_term`) and one is for indexable meta `wp_yoast_indexable_meta`. 

The migrations should be filled once. The table for primary term will be filled by selecting primary terms for taxonomies. The other two belongs to indexable actions like saving and editing posts/taxonomies/authors.

Most of the data will be added by using the metabox. For each type (post type, taxonomy, author) use the metabox to set the data. The social media data should be saved in the meta table. Please verify this works. 

* Check out the branch
* Run `composer install` to make sure the dependencies are prefixed
* Save posts/author/taxonomies and see the table being updated
* Check the attachment page as well. This should still work like before.

## Quality assurance


Fixes #8629
